LiteCore/tests/VectorQueryTest.cc

```diff
@@ -227,23 +227,160 @@ N_WAY_TEST_CASE_METHOD(SIFTVectorQueryTest, "Hybrid Vector Query", "[Query][.Vec
-        WHERE:    ['AND', ['VECTOR_MATCH()', 'vecIndex', ['$target']],
-                          ['=', 0, ['%', ['._sequence'], 100]] ],
-        WHAT:     [ ['._id'], ['AS', ['VECTOR_DISTANCE()', 'vecIndex'], 'distance'] ],
-        ORDER_BY: [ ['.distance'] ],
-        LIMIT:    10
-     }] )";
-    Retained<Query> query{store->compileQuery(json5(queryStr), QueryLanguage::kJSON)};
+        Log("---- Querying with $target = data");
+        Query::Options options = optionsWithTargetVector(kTargetVector, kData);
+        if ( result.size() == 0 ) {
+            result = collectResults(query->createEnumerator(&options));
+            checkExpectedResults(result,
+                                 {"rec-5300", "rec-4900", "rec-7100", "rec-3600", "rec-8700", "rec-8500", "rec-2400",
+                                  "rec-4700", "rec-4300", "rec-2600"},
+                                 {85776, 90431, 92142, 92629, 94598, 94989, 104787, 106750, 113260, 116129});
+        } else {
+            // Assert that the SQL++ query yields exactly same result as the JSON query.
+            auto result0 = collectResults(query->createEnumerator(&options));
+            CHECK(result0 == result);
+        }
+    }
+}
 
-    Log("---- Querying with $target = data");
-    Query::Options options = optionsWithTargetVector(kTargetVector, kData);
-    checkExpectedResults(query->createEnumerator(&options),
-                         {"rec-5300", "rec-4900", "rec-7100", "rec-3600", "rec-8700", "rec-8500", "rec-2400",
-                          "rec-4700", "rec-4300", "rec-2600"},
-                         {85776, 90431, 92142, 92629, 94598, 94989, 104787, 106750, 113260, 116129});
+N_WAY_TEST_CASE_METHOD(SIFTVectorQueryTest, "Hybrid vs plain Vector Query", "[Query][.VectorSearch]") {
+    readVectorDocs();
+    {
+        // Add some docs without vector data, to ensure that doesn't break indexing:
+        ExclusiveTransaction t(db);
+        writeMultipleTypeDocs(t);
+        t.commit();
+    }
+    createVectorIndex();
+
+    // Case 1: plain query because there is no additional term in the WHERE clause
+    // ===================
+
+    string queryStr = "SELECT META().id, VECTOR_DISTANCE(vecIndex) AS distance FROM "s + collectionName
+                      + " WHERE VECTOR_MATCH(vecIndex, $target) "
+                        " ORDER BY distance LIMIT 10";
+    Retained<Query> query{store->compileQuery(queryStr, QueryLanguage::kN1QL)};
+
+    Query::Options                        options = optionsWithTargetVector(kTargetVector, kData);
+    std::vector<std::pair<string, float>> result  = collectResults(query->createEnumerator(&options));
+
+    // The outer limit 10 is provided to VECTOR_MATCH as argument max_results
+    CHECK(result.size() == 10);
+
+
+    // Case 2: plain query because of explicitly assigned max_results to VECTOR_MATCH
+    // ===================
+
+    queryStr = "SELECT META().id, VECTOR_DISTANCE(vecIndex) AS distance FROM "s + collectionName
```

Comment:
Again, don't we test this already?

Reply:
Assert the equivalence for the plain query, whether use LIMIT or max_results.